### ISSUE TITLE
CT: do we really need to check that chemical masses do not exceed solubility?

### ISSUE THREAD

**Issue by vgherard** [erahumed/erahumed#143]
If the findings of #140 are *not* due to an error in units of measurement, it means that chemical masses are always way below solubility, in which case we can omit this step from the solution of the dynamical system.

This would be actually very convenient, because the system would then become linear, and it would be relatively easy to implement an efficient solution.

**Comment by vgherard** [comment]
Short answer: yes. 

For instance, for Cyhalofop-butyl that has the relatively low solubility of 0.44 ppm, the values of `r = density_in_water / solubility` get quite close to 1 (e.g. 90%). Even though these values never become exactly one (as would be the case if the solubility threshold was hit at some point), this means that it is not true that masses are *way below* solubility, and a minor change in other parts of the algorithm could make solubility become relevant.

Long story short: we need to keep solubility into the game.
